src/sparsezoo/analyze_v2/model_validators/distribution_analysis_model.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Optional, Union
+
+from pydantic import BaseModel, validator
+
+from sparsezoo.analyze_v2.model_validators.utils import type_validator
+
+
+class DistributionAnalysisModel(BaseModel):
+    counts: Optional[int]
+    mean: Optional[float]
+    median: Optional[float]
+    modes: Optional[List]
+    sum_val: Optional[float]
+    min_val: Optional[float]
+    max_val: Optional[float]
+    percentiles: Optional[Dict[float, float]]
+    std_dev: Optional[float]
+    skewness: Optional[float]
+    kurtosis: Optional[float]
+    entropy: Optional[float]
+    bin_width: Optional[float]
+    num_bins: Optional[int]
+    hist: Optional[List[float]]
+    bin_edges: Optional[List[float]]
+
+    @validator("*", pre=True)
```

Comment:
What is the role of the asterisk and pre=True in this validator method?

Reply:
"*" tells pydantic to run the validator for all values. 
The reason I do it for all is because yaml conversion will be messed up in `to_yaml()`, because some type might be numpy.generic.
 pre tells to run the validator before populating the values